### PR TITLE
#167190492 Implement Social Login

### PR DESCRIPTION
**Pull request by chialuka**
#### What does this PR do?
- Enables social login

#### Description of Task proposed in this pull request?
- Social login buttons were placed on the login page and wired to log the user in and then redirect them to their home page

#### How should this be manually tested (Quality Assurance)?
- Navigate to the login page of the application and click on any of the social icons to login with your social media account

#### What are the relevant pivotal tracker stories?
- [[167190492]](https://www.pivotaltracker.com/story/show/167190492)

#### Any background context you want to add (Operations Impact)?
- Run `npm install` and ensure to add the correct server URL for making API calls

#### What I have learned working on this feature: 
- I have learned how to implement social login from the backend to the frontend and how to interact with the Redux store

### CODE REVIEW

**Review by abdulfatai360 — APPROVED** [review]

**Review by Adekoreday — APPROVED** [review]

File: e2e-tests/index.js [hunk 1]
```diff
@@ -10,4 +10,18 @@ module.exports = {
       .assert.containsText('#main', 'Night Watch')
       .end();
   },
+
+  'Test social login: click social login': (client) => {
```
Comment (dinobi) [inline e2e-tests/index.js]: where's the assertion for this, and why is it in this file?

File: src/actionTypes/index.js [hunk 2]
```diff
@@ -2,3 +2,6 @@ export const SUCCESS = 'SUCCESS';
 export const REQUEST = 'REQUEST';
 export const FAILURE = 'FAILURE';
 export const GET_USER_PROFILE = 'GET_USER_PROFILE';
+export const FETCH_USER_HOMEPAGE_BEGIN = 'FETCH_USER_HOMEPAGE_BEGIN';
```
Comment (dinobi) [inline src/actionTypes/index.js]: why this action type named like this

File: src/actions/homePage.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {
+  FETCH_USER_HOMEPAGE_BEGIN,
+  FETCH_USER_HOMEPAGE_SUCCESS,
+  FETCH_USER_HOMEPAGE_ERROR
+} from '../actionTypes';
+import fetchData from '../helpers';
+
+const loadUserHomePage = (token, email) => async (dispatch) => {
```
Comment (dinobi) [inline src/actions/homePage.js]: please rename this function to what it's about

**Review by JuwonAbiola — APPROVED** [review]
Looks good to  me

**Pull request merged**